framework/src/meshgenerators/ParsedGenerateSideset.C

```diff
@@ -191,13 +200,29 @@ ParsedGenerateSideset::generate()
       if (_check_normal && std::abs(1.0 - _normal * normals[0]) > _variance)
         continue;
 
+      // check that boundary is within the list of included boundaries
+      if (_check_boundaries)
+      {
+        bool in_included_boundaries = false;
+        for (auto bid : restricted_boundary_ids)
+          if (boundary_info.has_boundary_id(elem, side, bid))
+            in_included_boundaries = true;
+        if (!in_included_boundaries)
+          continue;
+      }
+
       // check expression
       std::unique_ptr<Elem> curr_side = elem->side_ptr(side);
       _func_params[0] = curr_side->vertex_average()(0);
       _func_params[1] = curr_side->vertex_average()(1);
       _func_params[2] = curr_side->vertex_average()(2);
       if (evaluate(_func_F))
+      {
         boundary_info.add_side(elem, side, boundary_ids[0]);
+        // Create nodeset by default
+        for (auto & node : curr_side->node_ref_range())
+          boundary_info.add_node(&node, boundary_ids[0]);
```

Comment:
```suggestion
```
I don't like this very much. The class is suffixed with `Sideset`. We already build node lists from side lists by default anyway after mesh generation is all done. You can remove the line in this class where we set the name of the nodeset for more consistency